js/background.js

```diff
@@ -233,14 +233,13 @@
     }
 
     function onSentMessage(ev) {
-        var now = new Date().getTime();
         var data = ev.data;
 
         var message = new Whisper.Message({
             source         : textsecure.storage.user.getNumber(),
             sourceDevice   : data.device,
             sent_at        : data.timestamp,
-            received_at    : now,
+            received_at    : data.receivedAt || Date.now(),
```

Comment:
In this case, it looks like we're talking about when the message was **sent** but storing it in a prop called **received**At. Is that right? 

If I'm understanding, we have a similar field in iOS on which we just gave up and called `dateForSorting`, which is admittedly kind of an ugly name, but is less confusing in terms of self documenting code.

Also, when would `data.receivedAt` already be set - maybe on retry?